lib/puppet/provider/tcp_conn_validator/tcp_port.rb

```diff
@@ -0,0 +1,47 @@
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__),"..","..",".."))
+require 'puppet_x/puppet-community/tcp_validator'
+
+# This file contains a provider for the resource type `tcp_conn_validator`,
+# which validates the TCP connection.
+
+Puppet::Type.type(:tcp_conn_validator).provide(:tcp_port) do
+  desc "A provider for the resource type `tcp_conn_validator`,
+        which validates the tcp connection."
+
+  def exists?
+    start_time = Time.now
+    timeout = resource[:timeout]
+
+    success = validator.attempt_connection
+
+    while success == false && ((Time.now - start_time) < timeout)
+      Puppet.debug("Failed to connect to the server; sleeping 4 seconds before retry")
+      sleep 4
```

Comment:
This surprised me. The default wait of 15 seconds doesn't evenly divide into four second blocks. I'd expect a 1 second sleep, and I could see value in exposing this as a parameter.